esmvaltool/namelists/namelist_preprocessor_derive_test.yml

```diff
@@ -0,0 +1,27 @@
+---
+
+models:
+#  - {model: CESM1-WACCM,    project: CMIP5, mip: Amon, exp: historical, ensemble: r2i1p1, start_year: 1997, end_year: 2005}
+#  - {model: CNRM-CM5,       project: CMIP5, mip: Amon, exp: historical, ensemble: r1i1p1, start_year: 1997, end_year: 2005}
+  - {model: GFDL-CM3,       project: CMIP5, mip: Amon, exp: historical, ensemble: r1i1p1, start_year: 1997, end_year: 2005}
+  - {model: GISS-E2-H,      project: CMIP5, mip: Amon, exp: historical, ensemble: r1i1p2, start_year: 1997, end_year: 2005}
+  - {model: GISS-E2-R,      project: CMIP5, mip: Amon, exp: historical, ensemble: r1i1p2, start_year: 1997, end_year: 2005}
+#   - {model: MIROC-ESM-CHEM, project: CMIP5, mip: Amon, exp: historical, ensemble: r1i1p1, start_year: 1997, end_year: 2005}
+  - {model: ESACCI-OZONE,   project: OBS,   tier: 2,   type: sat,       version: L3,      start_year: 1997, end_year: 2005}
+#   - {model: NIWA,           project: OBS,   tier: 3,   type: reanaly,   version: 1,       start_year: 1997, end_year: 2005}
+
+preprocessors:
+
+  preprocessor:
+    derive: true
```

Comment:
Why do we need this switch?

Reply:
I'll have a look, I probably made a mistake there.

Reply:
My apologies, I didn't look carefully at your question.

The preprocessor needs to know that it has to derive a variable. We can either attempt to derive a variable automatically when no input files are found, or be more explicit and specify it in the namelist somewhere. I liked the idea of being explicit about what we do, so I specified it in the namelist. For me, the logical place to do so is the preprocessor settings, because it is a preprocessor function. However, if you want things done differently, let me know and i can change it.

Reply:
I agree, this should be explicitly visible in the namelist.
I find `variable` a more appropriate place for this settings for several reasons:
- it is a variable-dependent setting, like `reference_model` for example;
- it mimics somehow the behaviour of v1.0, where this is set in `variable_defs/`, so it might be more intuitive for old users;
- putting it in the preprocessor would hamper the use of the same preprocessor for multiple variables (derived and non-derived) and a double copy of the preprocessor just differing in `derive` would have to be created instead.

Reply:
Ok, I've changed it, though I personally find it confusing to configure preprocessor functions in the variables section instead of in the preprocessor section.